schema/Readme.md

```diff
@@ -920,73 +932,195 @@ model User {
 model Customer {}
 ```
 
-## Type Definition
+## Type Mapping
 
 > ⚠ This is not implemented yet.
 
-Type definitions can be used to consolidate various type specifications into one type.
+Type mappings bridge Prisma's data types with the datasource's data types. Type mappings also provide a way to choose the transportation format.
 
-```groovy
-type Numeric {
-  precision Int @bound(gte: 1, lte: 131072)
-  scale Int @bound(gte: 0, lte: 16383)
-}
+### Terminology
 
-model User {
-  id       Int      @id
-  weight   Numeric(precision: 5, scale: 20)
-}
+#### Application Schema
+
+The Application Schema is typically defined by your `schema.prisma`. This is where your application's models and relationships will live.
+
+#### Connector Schema
+
+The Connector Schema is a virtual file that is generated by the connectors. This file will be named after the connector, so if the connector is `postgresql`,
+then the virtual file will be named `postgresql.schema`. This file contains Connector Type Definitions.
+
+#### Connector Type Definitions
+
+Connector types definitions declare a connector-specific type and how it maps to the underlying datasource.
+
+```groovy
+type BigInt @raw("bigint") @transport("int64")
 ```
 
-You can attach any field attribute to a type definition.
+Connector type definitions have 2 attributes:
 
-### Type Definitions provided by Connectors
+- `@raw(_ name: String)` defines the type we map to in the datasource.
+- `@transport(_ name: String)` defines how we transport this type over the network
```

Comment:
I think @schickling has suggested that we use the Prisma core types for now, but graduate to something like Protobufs in the future. I personally would very much like to push this out into the future since I don't think it matters right now.